src/app/components/v2/map/map.js

```diff
@@ -52,18 +53,33 @@ class CesiumComponent extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { onTick, lockNavigation = false, rotate, coordinates, coordinatesOptions, camera } = this.props;
+    const {
+      onTick,
+      lockNavigation = false,
+      rotate,
+      coordinates,
+      coordinatesOptions,
+      camera,
+      terrainMode,
+      latLng,
+      terrainCameraOffset
+    } = this.props;
     if (this.viewer) {
       if (!this.rotating && rotate) this.addRotation();
       if (this.rotating && !rotate) this.removeRotation();
       if (onTick && !this.ticking) this.addTick();
       if (camera && prevProps.camera !== camera) this.setCamera();
       if (lockNavigation) disablePanning(this.viewer);
       if (
-        coordinates && (prevProps.coordinates !== coordinates || prevProps.coordinatesOptions !== coordinatesOptions)
+        coordinates &&
+          !terrainMode &&
+          (prevProps.coordinates !== coordinates || prevProps.coordinatesOptions !== coordinatesOptions)
       ) {
         this.setCoordinates();
       }
+      if (terrainMode && prevProps.coordinates === coordinates) {
```

Comment:
Shouldn't this be the latLng property? we are not using coordinates for terrainMode.

Reply:
Not completely, we are coupling the feature of changing coordinates to the lat&Lng one, so why not check if the latLng is different instead?

Reply:
We are going to remove `latLng` from the URL after escaping `terrainMode`?

Reply:
⚡️ addressed 
Much better this way.
Thank you!!

Reply:
Is kinda ugly, but being an object have to compare its properties.